src/main/java/com/evoter/candidate/service/CandidateServiceImpl.java

```diff
@@ -24,33 +31,52 @@ public Candidate createCandidate(CreateUpdateCandidateRequestDto requestDto) {
         log.info("Request to create a candidate with payload={}", requestDto);
 
         // validate that candidate details are not null
+        if(requestDto ==null) throw new GeneralException(ResponseCodeAndMessage.INVALID_JSON_REQUEST_DATA_90
+                .responseCode,"request cannot be null");
 
         // check if candidate already exist
-
-        //save
         Candidate candidate = new Candidate();
         candidate.setName(requestDto.getName());
         candidate.setAge(requestDto.getAge());
         candidate.setSex(requestDto.getSex());
         candidate.setPartyId(requestDto.getPartyId());
         candidate.setPollTypeId(requestDto.getPollTypeId());
+        if(candidateExists(candidate)) throw new GeneralException(ResponseCodeAndMessage.ALREADY_EXIST_86.
+                responseCode,"Candidates already exists");
 
+        //save
         return candidateRepository.save(candidate);
     }
 
+    public boolean candidateExists(Candidate candidate) {
```

Comment:
using TypeQuery in this case is not a good thing, instead create a method in the repository called isExistByName(String name) that checks if that candidate name exist.